File: MVCProject/app/src/main/java/com/example/mvc_project/db/ProductDAO.java
package com.example.mvc_project.db;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import com.example.mvc_project.model.Product;

import java.util.List;

@Dao
public interface ProductDAO {
    @Query("SELECT * From products")
    LiveData<List<Product>> getFavProducts();
    @Insert(onConflict = OnConflictStrategy.IGNORE)
    void insertProduct(Product product);
    @Delete
    void deleteMovie (Product product);
}

File: MVVM/app/src/main/java/com/example/mvvm/database/ProductDao.kt
package com.example.mvvm.database

import androidx.lifecycle.LiveData
import androidx.room.*
import com.example.mvvm.model.Product

@Dao
interface ProductDao {
    @Query("SELECT * From products")
    suspend fun getFavProducts(): List<Product>
    @Insert(onConflict = OnConflictStrategy.IGNORE)
   suspend fun insertProduct(product: Product)
    @Delete
    suspend fun deleteMovie(product: Product?)

}